internal/dag/gatewayapi_processor.go

```diff
@@ -276,6 +277,84 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 		}()
 	}
 
+	// Compute each GRPCRoute for each Listener that it potentially attaches to.
+	for _, grpcRoute := range p.source.grpcroutes {
+		func() {
```

Comment:
I have a WIP branch for extracting a lot of this code that's repeated across all of the route types into a helper, but can save that for a subsequent refactoring PR since this one is already large.